server/conn.go

```diff
@@ -402,31 +406,37 @@ func (cc *clientConn) Run() {
 			buf = buf[:stackSize]
 			log.Errorf("lastCmd %s, %v, %s", cc.lastCmd, r, buf)
 		}
-		err := cc.Close()
-		terror.Log(errors.Trace(err))
+		if !closedOutside {
+			err := cc.Close()
+			terror.Log(errors.Trace(err))
+		}
 	}()
 
-	for !cc.killed {
-		atomic.StoreInt32(&cc.active, 0)
+	cc.status = connStatusDispatching
+	for {
+		if atomic.CompareAndSwapInt32(&cc.status, connStatusDispatching, connStatusReading) == false {
+			closedOutside = true
```

Comment:
The server doesn't close a dispatching connection.

Reply:
The server doesn't close a dispatching connection, but it would set the status to Shutdown to notify.
So it's checking the notification here.